### ISSUE TITLE
File visibility with Google cloud

### ISSUE THREAD

**Issue by LarBearrr** [talvbansal/media-manager#91]
Any ideas on how I can set the visibility of my files/folders on Google Cloud Storage as they are uploaded with the media manager?

I have set the `acccess` key in config/media-manager.php to `public` but it does not seem to be doing anything.

So far I have been able to implement Google Cloud Storage using the [superbalist/laravel-google-cloud-storage](https://github.com/Superbalist/laravel-google-cloud-storage) package. I can verify that it is working by uploading a file, creating folders, etc. Furthermore, the directory structure in the media manager mirrors that of my Google Bucket. However, none of the uploaded files are viewable because their visibility has not been set public. 

I can do a test with the following line `Storage::disk('gcs')->setVisibility($location, 'public');` and see that it does set the `$location` in the bucket to public. 

Are there any suggestions on using another cloud driver, like GCS, and changing the visibility of uploaded files?

Thank you.

**Comment by talvbansal** [comment]
Hi @LarBearrr!

Thanks for checking the package out. Sorry for the delay in responding I'm currently out travelling at the moment. 

I've not worked with google cloud storage so I'm not familiar with the details of it but i will try my very best to look into it for you. If this is more time sensitive perhaps we can look at working on this over zoom / skype / slack.

Thanks

Talv

**Comment by LarBearrr** [comment]
So I did some more digging and was able to see where I can accomplish this task. If you look [here ](https://github.com/talvbansal/media-manager/blob/ea555c4219cad9581edd3e1ef5673f9ae5bd11a6/src/Services/MediaManager.php#L435) you will see that the storeAs() method is taking 4 parameters, when really it only accept 3: the path, filename, and an array of options. 

This being said, I created an array of options and included the visibility set to public in that array and then passed in the options to the storeAs() method:


```
....
$options = [
                'disk' => $this->diskName,
                'visibility' => 'public'
            ];
            if (!$file->storeAs($path, $fileName, $options)) {
                $this->errors[] = trans('media::messages.upload_error', ['entity' => $fileName]);

                return $uploaded;
            }
....
```

This works for my purposes, however I am not sure if it will work with other providers or using the local disk. Something to think about. Let me know if you have had any luck on your end or what your thoughts are.

Thanks again!
